src/database.cc

```diff
@@ -115,7 +115,13 @@ Database::Database(const Napi::CallbackInfo& info) : Napi::ObjectWrap<Database>(
 
     int mode;
     if (info.Length() >= pos && info[pos].IsNumber()) {
-        mode = info[pos++].As<Napi::Number>().Int32Value();
+        double orig_val = info[pos].As<Napi::Number>().DoubleValue();
+        double int_val = (double)info[pos].As<Napi::Number>().Int32Value();
+        if (orig_val == int_val) {
```

Comment:
Did you check if this does the right thing for cases like Number.NEGATIVE_INFINITY
Number.POSITIVE_INFINITY etc?